pyiron_base/project/path.py

```diff
@@ -391,10 +391,9 @@ def _create_path(self, path, rel_path=None):
         if rel_path:
             rel_path = self._windows_path_to_unix_path(rel_path)
             path = posixpath.join(path, rel_path)
-        try:
-            os.makedirs(path)
-        except os.error:
-            pass
+        os.makedirs(path, exist_ok=True)
+        if not os.path.exists(path):
+            raise os.error("pyiron cannot create directory.")
```

Comment:
Do these lines actually have an effect? Either `os.makedirs` succeeds and doesn't raise an error, but then path exists, or it fails and the path doesn't exist, but it'll raise an error, right?

Reply:
No, that is the tricky part. When you are out of space - reached the limit of your quota - then `os.makedir` raises an `OsError` but Python thinks this means the folder already exists and continues. We just had this case recently and it is very tricky to find, as you create a project like `Project('test')` and no folder is created but also no error appears. Only later when you try to write into the project the script fails. Unfortunately, with a tool like pyiron it is rather easy to reach the limits of your quota, so I presume we might have people with similar issues soon.

Reply:
Oh, I see.  I would add a comment to this effect, though.  We can also think about raising this as a bug with python itself.

Reply:
I like that idea, let's see what they respond https://github.com/python/cpython/issues/101321

Reply:
I had to look at the underlying code to realize it was not python's fault. The issue was on our side, as we had: 
```
         try:
             os.makedirs(path)
         except os.error:
             pass
```
This could be resolved by using `os.makedirs(path, exists_ok=True)`.